packages/core/src/electron-common/messaging/electron-messages.ts

```diff
@@ -51,3 +53,70 @@ export interface CloseRequestArguments {
     cancelChannel: string;
     reason: StopReason;
 }
+
+/**
+ * Request is sent by the render to the main to rebuild and set the menu.
+ */
+export namespace SetMenu {
+    export const Signal = 'setMenu';
+    export interface Params {
+        readonly template: MenuItemConstructorOptions[] | undefined;
+    }
+}
+/**
+ * Request sent by the render to the main to build and show the context menu.
+ */
+export namespace ShowContextMenu {
+    export const Signal = 'showContextMenu';
+    export interface Params {
+        readonly template: MenuItemConstructorOptions[];
+        /** Must be an integer. */
+        readonly x: number;
+        /** Must be an integer. */
+        readonly y: number;
+        readonly contextMenuId: string;
+    }
+};
+/**
+ * Sent by the renderer to main to programmatically close an open context menu identified by the unique `contextMenuId`.
+ * This request might be ignored by the recipient if no open context menu exists with the ID. For example, the context menu was closed by user clicking away.
+ */
+export namespace CloseContextMenu {
+    export const Signal = 'closeContextMenu';
+    export interface Params {
+        readonly contextMenuId: string;
+    }
+}
+/**
+ * Event sent by the main to the render when a context menu was closed.
+ */
+export namespace ContextMenuDidClose {
+    export const Signal = 'contextMenuDidClose';
+    export interface Params {
+        readonly contextMenuId: string;
+    }
+}
+/**
+ * Event sent by the main to the renderer when one of the menu items was clicked.
+ */
+export namespace MenuItemDidClick {
+    export const Signal = 'menuItemDidClick';
+    export interface Params {
+        readonly commandId: string;
+        /**
+         * The arguments will be serialized with the [_Structured Clone Algorithm_](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm).
+         * Sending Functions, Promises, Symbols, DOM or special electron objects, WeakMaps, or WeakSets will throw an exception.
+         */
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        readonly args?: any[];
```

Comment:
The renderer should always already know the arguments that should be passed to command handlers, so I'm not sure we ever need to send complex data between the renderer and the main process. The main process only needs to know enough to render the menu - the renderer has all the information it needs to know how to handle a click when the main process informs it that one has occurred.

Reply:
I agree.